pkg/retry/failure_retryer.go

```diff
@@ -295,10 +295,12 @@ func (r *failureRetryer) Dump() map[string]interface{} {
 
 func (r *failureRetryer) setSpecifiedResultRetryIfNeeded(rr *ShouldResultRetry) {
 	if rr != nil {
+		// save the object specified by client.WithSpecifiedResultRetry(..)
 		r.specifiedResultRetry = rr
 	}
```

Comment:
Old implemetation is inconsistent: a `FailurePolicy` with nil `ShouldResultRetry` will be overwritten, while non-nil `ShouldResultRetry` will not.

Reply:
In the new impl, `client.WithSpecifiedResultRetry` always has the highest priority.